src/Raven.Server/Documents/ResourceCache.cs

```diff
@@ -165,20 +165,10 @@ public IDisposable RemoveLockAndReturn(string databaseName, Action<TResource> on
 
                 task.IgnoreUnobservedExceptions();
 
-                bool found = false;
-                while (found == false)
+                if (_caseInsensitive.TryGetValue(databaseName, out current) == false)
                 {
-                    found = _caseInsensitive.TryGetValue(databaseName, out current);
-                    if (found == false)
-                    {
-                        resource = default(TResource);
-                        if (_caseInsensitive.TryAdd(databaseName, task) == false)
```

Comment:
What is the result of not adding it at all?

Reply:
Not sure why we added that in the first place...